### PR TITLE
feature(tests): for buttons

### PR DESCRIPTION
**Pull request by ameli90**
for Button.vue, ButtonCheckbox.vue, ButtonRadio.vue

### CODE REVIEW

File: test/unit/Button.spec.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/* global describe, it, expect */
+
+import Vue from 'vue/dist/vue.js'
+import VkButton from '../../src/lib/Button.vue'
+
+describe('Button.vue state classes', () => {
+  it('should be active', () => {
+    const vm = new Vue({
+      template: '<div><vk-button active>My text button</vk-button></div>',
```
Comment (miljan-aleksic) [inline test/unit/Button.spec.js]: Change `My text button` to `Button` on all examples.

File: test/unit/Button.spec.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      components: {
+        VkButton: VkButton
+      }
+    }).$mount()
+    expect(vm.$el.querySelector('button').classList.contains('uk-active')).toBeTruthy()
+  })
+  it('should be disabled', () => {
+    const vm = new Vue({
+      template: '<div><vk-button disabled>My text button</vk-button></div>',
```
Comment (miljan-aleksic) [inline test/unit/Button.spec.js]: Is the `div` wrapper necessary?

File: test/unit/Button.spec.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      components: {
+        VkButton: VkButton
+      }
+    }).$mount()
+    expect(vm.$el.querySelector('button').hasAttribute('disabled')).toBeTruthy()
```
Comment (miljan-aleksic) [inline test/unit/Button.spec.js]: Can the `vm.$el.querySelector('button')` query be saved and rused?

File: test/unit/Button.spec.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+/* global describe, it, expect */
+
+import Vue from 'vue/dist/vue.js'
+import VkButton from '../../src/lib/Button.vue'
+
+describe('Button.vue state classes', () => {
```
Comment (miljan-aleksic) [inline test/unit/Button.spec.js]: Change it to `VkButton state

File: test/unit/Button.spec.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      template: '<div><vk-button disabled>My text button</vk-button></div>',
+      components: {
+        VkButton: VkButton
+      }
+    }).$mount()
+    expect(vm.$el.querySelector('button').hasAttribute('disabled')).toBeTruthy()
+  })
+})
+
+describe('Button.vue size', () => {
```
Comment (miljan-aleksic) [inline test/unit/Button.spec.js]: `VkButton size`

File: test/unit/Button.spec.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+import Vue from 'vue/dist/vue.js'
+import VkButton from '../../src/lib/Button.vue'
+
+describe('Button.vue state classes', () => {
+  it('should be active', () => {
+    const vm = new Vue({
+      template: '<div><vk-button active>My text button</vk-button></div>',
+      components: {
+        VkButton: VkButton
```
Comment (miljan-aleksic) [inline test/unit/Button.spec.js]: Try registering the components globally.

File: test/unit/ButtonCheckbox.spec.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
+/* global describe, it, expect, beforeEach */
+
+import Vue from 'vue/dist/vue.js'
+import VkButton from '../../src/lib/Button.vue'
+import VkButtonCheckbox from '../../src/lib/ButtonCheckbox.vue'
+
+describe('ButtonCheckbox.vue', () => {
+  var vm
```
Comment (miljan-aleksic) [inline test/unit/ButtonCheckbox.spec.js]: Use `const`

File: test/unit/ButtonCheckbox.spec.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    vm = new Vue({
+      data: {
+        values: [],
+        group: false
+      },
+      components: {
+        VkButton,
+        VkButtonCheckbox
+      },
+      template: '<vk-button-checkbox :group="group" v-model="values" @change="values = arguments[0]">' +
```
Comment (miljan-aleksic) [inline test/unit/ButtonCheckbox.spec.js]: Use ES6 template literals for multiline strings.

File: build/karma.conf.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
+// we can just use the exact same webpack config by requiring it
+// but make sure to delete the normal entry
+var webpackConf = require('./webpack.base.conf')
+delete webpackConf.entry
+
+module.exports = function (config) {
```
Comment (miljan-aleksic) [inline build/karma.conf.js]: Add more tests:
- Tag test making sure is `button`
- Aria type test

Comment (ameli90) [inline build/karma.conf.js]: - added tests for element tag name
- for aria checked attribute there is a test https://github.com/vuikit/vuikit/pull/18/files#diff-b143da5a85bf7b3ed6f655995fd5e983R37, do you mean something else?

File: test/unit/ButtonCheckbox.spec.js (hunk 7 above)
Comment (ameli90) [inline test/unit/ButtonCheckbox.spec.js]: cannot make it as a `const` here, as need to define it globally for all tests, and init `vm` in `beforeEach()`

Comment (miljan-aleksic) [inline test/unit/ButtonCheckbox.spec.js]: In that case use 'let', avoid using 'var'.

**Comment by miljan-aleksic** [comment]
Update your local copy witht the latest updates as the PR is trying to revert them now.

**Pull request closed without merging**